arch/arm/core/cortex_m/prep_c.c

```diff
@@ -24,6 +24,45 @@
 #include <arch/arm/cortex_m/cmsis.h>
 #include <string.h>
 
+extern uint32_t _interrupt_stack;
+
+static inline void switch_sp_to_psp(void)
+{
+	__set_CONTROL(__get_CONTROL() | CONTROL_SPSEL_Msk);
+	/* Synchronous barrier to make psp new SP effective immediately */
```

Comment:
I would copy the previous explanatory text here

Reply:
ok.